src/bpfbytecode.cpp

```diff
@@ -44,6 +49,54 @@ const std::vector<uint8_t> &BpfBytecode::getSection(
   return sections_.at(name);
 }
 
+BpfProgram &BpfBytecode::getProgramForProbe(const Probe &probe)
```

Comment:
I think that it requires a bit of `const_cast`-magic:
```
const BpfProgram &BpfBytecode::getProgramForProbe(const Probe &probe) const
{
   ... original code ...
}

BpfProgram &BpfBytecode::getProgramForProbe(const Probe &probe)
{
  return const_cast<BpfProgram &>(
      const_cast<const BpfBytecode *>(this)->getProgramForProbe(probe));
}
```
but since the non-const version will go away soon, I'm ok with doing this as it makes clear that `AttachedProbe` is not changing `BpfProgram`.

Reply:
But if the non-const one is going away soon then I guess it's not important